### DISCUSSION TITLE
60

### DISCUSSION THREAD

**Discussion by jojoldu** [jojoldu/blog-comments#720] [Comments]
http://jojoldu.tistory.com/60

**Comment by jojoldu** [comment]
작성자: 아루아빠

좋은 내용 잘 봤습니다. 감사합니다.

**Comment by jojoldu** [comment]
작성자: ela

좋은 내용이네요!

**Comment by jojoldu** [comment]
작성자: 초보

안녕하세요. 좋은 글 감사합니다.
한가지 궁금한 점이 있는데, Nginx를 프록시서버로만 이용한다고 하셨는데,
Static 파일들(ex:css, js 등..)을 Nginx에서 바로 처리하지 않고,
톰캣으로 넘겨서 처리하는 특별한 이유가 있나요??

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

음? 제가 댓글을 잘못이해한건지..
본문을 보면 정적파일들은 다 Nginx에서 캐시한다고 작성했어서요!
정적 << 으로 검색하면 해당 내용이 본문에 포함되있습니다!

**Comment by jojoldu** [comment]
작성자: 초보

아.. 제가 질문을 명확하게 하지못했네요ㅠ 죄송합니다ㅎㅎ 말씀해주신것과같이 Static파일을 Nginx에서 캐시한다고 하셨는데,  제가 본문 내용을 제대로 이해하지못했을수도 있지만 제가 이해한 바로는, 캐시가 되어있으면 캐시된 데이터를 내려주고, 캐시가 안되었을경우(ex:최초 1회? 또는 캐시 만료 이후 등..) Tomcat에서 Static파일이 처리되는것으로 생각되는데, Tomcat으로 넘기지않고, 
 Static파일은 말 그대로 정적파일이니 WAS가 필요없으므로 경로 잡아서 바로 Nginx에서 처리해줄 수 있지 않을까.. 생각되어서 질문드렸습니다ㅎㅎㅎ

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

아! 넵넵
외장톰캣을 이용한 기존의 스프링 프로젝트나 최근의 스프링부트를 사용한 방법이나 모두 프로젝트 내부에 스태틱 파일을 포함해서 배포하고 있습니다.
S3와 같이 공용 스태틱 저장소를 사용하지 않고, IDC를 이용한 30대의 물리서버를 사용하다보니 각 서버의 Nginx가 스태틱 파일을 별도로 관리하도록 구축하는게 불편한게 많았습니다.
그래서 프로젝트에 정적파일을 포함시켜서 배포한뒤,  Nginx에서 첫 요청이후로 전부 캐시하도록 하였습니다.
(아마 스프링 프로젝트에서 가장 많이들 사용하는 방법이기도 할것 같습니다) 

만약 Nginx 서버와 톰캣 서버가 분리되어있고,
AWS 같은 클라우드 서비스의 S3를 이용하고 있었다면 정적 파일은 모두 S3에 올려서 사용했을것 같습니다!

**Comment by JennaGil** [comment]
안녕하세요 
nginx 의 프록시 서버로 사용하고 정적데이터의 캐싱방법에 대해서 고민하고 있는 개발자입니다. 
한가지 궁금한점이 있습니다. 
실무에서는 nginx는 여러대를 두고 로드밸런싱해서 사용할텐데, 캐싱된 데이터를 삭제하려고 할때 각 서버마다 모두 들어가서 삭제할수는 없을텐데 어떤 방법을 취하셨는지 알 수 있을까요? 혹시 캐시 저장공간은 각 nginx 서버에서 마운트해서 사용하는지요. 정리해주신글 잘 봤습니다~

**Comment by jojoldu** [comment]
@JaneKwon82 안녕하세요?
최근엔 Nginx로 정적파일 캐시를 쓰지 않습니다.
CloudFront나 Cloudflare 등 CDN을 통해서 처리하고 있어, 분산된 서버의 캐시 파일을 별도로 관리하진 않습니다.
만약 갱신된 정적파일 호출이 필요하면, 뒤에 파라미터를 붙여서 호출하면 되는거라 크게 신경쓰지 않습니다.

**Comment by ljhyeok** [comment]
헐 상세한 포스팅 감사합니다!
aws elastic beanstalk으로 ec2 두 대 쓰고 
각 ec2 에 nginx 프록시 서버 쓰고 있었는데, 배포할때만 동일한 문제 겪었어요.....
nginx cache 파일 때문에 그렇군요....
배포할때마다 캐시 파일 날리는 명령어만 추가하면 제가 갖고 있는 문제는 해결 될 것 같아요
소중한 포스팅 감사합니다

**Comment by woosungchoi** [comment]
좋은 정보 감사합니다.

**Comment by jobmania** [comment]
ㅋㅋㅋ 내용도 좋고 짤두 재밋네영

**Comment by mingkyme-xdn** [comment]
cache_key를 조절하여 해결하는 방법도 있겠네요
